euphonic/design/05_powder_averaging_discussion.md

```diff
@@ -0,0 +1,318 @@
+-   Consider the topology/features of the space; there are more
+    features intersecting a spherical shell at large *q* than a
+    smaller shell at low *q*. This implies that a greater number of
+    samples is needed as we go further out in *q*.
+
+### How can we do this efficiently?
+
+-   Use symmetry!
+    - The irreducible wedge of the Brillouin zone can be
+      drastically smaller than the full space while containing all
+      the unique information.
+    - Don't try too hard to make origin shifts that squeeze a bit
+      more efficiency from symmetry operations; these benefits
+      diminish as the number of samples increases.
+-   Use caching
+    - The *q*-range in INS measurements passes through many cells
+      of the reciprocal lattice. If possible, re-use computed
+      frequency/qpts in extended zone scheme.
+    - Brille solves some of these problems by transparently
+      re-using appropriate data.
+    - Would be nice to intentionally select long-range *q-points*
+      that already have calculated frequencies so that only S
+      needs recalculating. This might not be commensurate with the
+      desired *q* bins, however.
+
+### How can we check the convergence? Is a "fire-and-forget" approach feasible?
+
+- This could be a substantial project in its own right. To progress
+  the project we should get a baseline method in place and then
+  experiment from there, rather than designing everything in theory
+  first.
+- Fundamentally the goal of the sampling exercise is to perform a
+  numerical integration. In the numerical methods field there has been
+  a lot of work on integration problems, including methods which use
+  adaptive step sizes to obtain a consistent uncertainty when
+  integrating arbitrarily complex functions.
+- This function is not arbitrary and there should be some useful
+  information lurking in the force constants data.
+
+
+### Is the same sampling mesh suitable for the **A** matrix used in Debye-Waller calculation, or should this be a separate q-mesh?
+
+- This point was not discussed in the meeting. It's worth observing
+  that DW should only need sampling within a single Brillouin zone,
+  while S reaches out further in reciprocal space. We'll probably stick with
+  Monkhorst-Pack grids initially.
+
+
+### Can we use statistical methods to quantify uncertainty in the convergence?
+
+- Statistical methods, propogation of error approaches could provide
+  some useful tools
+- Experimentally we can examine uncertainty by comparing different
+  detector contributions to the same bin
+- We should be able to somehow examine the changing variance in S as more
+  **q**-points are sampled.
+
+<a id="org72cee07"></a>
+
+## Existing approaches
+
+
+<a id="orgce0cfe3"></a>
+
+### Random sampling
+
+-   This has been mentioned in PACE meetings. How much has it been
+    used?
+    - Random sampling on spherical shells seems more common that fully
+      random sampling in **q**-space.
```

Comment:
An example is found in `spinw`: https://github.com/SpinW/spinw/blob/master/swfiles/%40spinw/powspec.m#L291 which does random sampling on specified abs(q) shells.